3.5/runtime/windowsservercore-2004/Dockerfile

```diff
@@ -0,0 +1,24 @@
+# escape=`
+
+FROM mcr.microsoft.com/windows/servercore:2004
+
+# Enable detection of running in a container
+ENV DOTNET_RUNNING_IN_CONTAINER=true `
+    # Ngen workaround: https://github.com/microsoft/dotnet-framework-docker/issues/231
+    COMPLUS_NGenProtectedProcess_FeatureEnabled=0
+
+# Install .NET Fx 3.5
+RUN curl -fSLo microsoft-windows-netfx3.zip https://dotnetbinaries.blob.core.windows.net/dockerassets/microsoft-windows-netfx3-2004.zip `
+    && tar -zxf microsoft-windows-netfx3.zip `
+    && del /F /Q microsoft-windows-netfx3.zip `
+    && DISM /Online /Quiet /Add-Package /PackagePath:.\microsoft-windows-netfx3-ondemand-package~31bf3856ad364e35~amd64~~.cab `
+    && del microsoft-windows-netfx3-ondemand-package~31bf3856ad364e35~amd64~~.cab `
+    && powershell Remove-Item -Force -Recurse ${Env:TEMP}\* `
+    && C:\Windows\Microsoft.NET\Framework64\v4.0.30319\ngen.exe update `
```

Comment:
It looks like the ngen layer that exists in the pre-2004 images was consolidated.  What is your plan going forward for applying the LCU?  Do you plan to do that in the same layer or a separate layer like it currently is?  I ask because this ngen should only happen after the LCU is applied.  

I think there is value in using consistent patterns across the image variants.  With this change there are inconsistencies in the runtime, aspnet, and sdk images in regards to running ngen update.

Reply:
Updated to make this consistent.